MapleServer2/PacketHandlers/Game/RockPaperScissorsHandler.cs

```diff
@@ -0,0 +1,320 @@
+        {
+            OpenMatch = 0x00,
+            RequestMatch = 0x01,
+            ConfirmMatch = 0x02,
+            DenyMatch = 0x03,
+            CancelRequestMatch = 0x05,
+            SelectRPSChoice = 0x07,
+            ClaimReward = 0x0A,
+            ConfirmMatch2 = 0x0C,
+        }
+
+        private enum MicroGameError : byte
+        {
+            OtherPlayerCannotPlayRightNow = 0x0,
+            CannotPlayInThisMap = 0x1,
+            FailedToStart = 0x4,
+        }
+
+        public override void Handle(GameSession session, PacketReader packet)
+        {
+            MicroGameMode mode = (MicroGameMode) packet.ReadShort();
+
+            switch (mode)
+            {
+                case MicroGameMode.OpenMatch:
+                    HandlOpenMatch(session);
+                    break;
+                case MicroGameMode.RequestMatch:
+                    HandleRequestMatch(session, packet);
+                    break;
+                case MicroGameMode.ConfirmMatch:
+                    HandleConfirmMatch(session, packet);
+                    break;
+                case MicroGameMode.DenyMatch:
+                    HandleDenyMatch(session, packet);
+                    break;
+                case MicroGameMode.CancelRequestMatch:
+                    HandleCancelRequestMatch(session, packet);
+                    break;
+                case MicroGameMode.SelectRPSChoice:
+                    HandleSelectRPSChoice(session, packet);
+                    break;
+                case MicroGameMode.ClaimReward:
+                    HandleClaimReward(session, packet);
+                    break;
+                case MicroGameMode.ConfirmMatch2:
+                    HandleConfirmMatch2(session, packet);
+                    break;
+                default:
+                    IPacketHandler<GameSession>.LogUnknownMode(mode);
+                    break;
+            }
+        }
+
+        private static void HandlOpenMatch(GameSession session)
+        {
+            RPS rpsEvent = DatabaseManager.Events.FindRockPaperScissorsEvent();
+            if (rpsEvent is null)
+            {
+                return;
+            }
+
+            if (!session.Player.Inventory.Items.Values.Any(x => x.Id == rpsEvent.VoucherId))
+            {
+                // TODO: Find correct packet to let player know they don't have a voucher
+                session.Send(NoticePacket.Notice("You must have a Rock Papers Scissors Play Ticket", NoticeType.ChatAndFastText));
+                return;
+            }
+
+            session.Send(RockPaperScissorsPacket.Open());
+        }
+
+        private static void HandleRequestMatch(GameSession session, PacketReader packet)
+        {
+            long characterId = packet.ReadLong();
+
+            Player otherPlayer = session.FieldManager.State.Players.FirstOrDefault(x => x.Value.Value.CharacterId == characterId).Value.Value;
+            if (otherPlayer == null)
```

Comment:
Without the '?' it'll crash if the character doesn't exist when trying to get `.Value`. Same thing for the other functions which use **otherPlayer**
```suggestion
            Player otherPlayer = session.FieldManager.State.Players.Values.FirstOrDefault(x => x.Value.CharacterId == characterId)?.Value;
            if (otherPlayer is null)
```